controller/admin_transactions_controller.php

```diff
@@ -395,6 +483,96 @@ public function get_valid_offset()
 		return ($this->last_page_offset) ? (int) $this->last_page_offset : 0;
 	}
 
+	/**
+	 * Prepare data array() before send it to $entity
+	 *
+	 * @param array $transaction_data
+	 *
+	 * @return array
+	 * @throws transaction_exception
+	 */
+	private function build_data_ary($transaction_data)
+	{
+		$errors = array();
+
+		if ($this->request->is_set('u' && $transaction_data['MT_USERNAME'] === ''))
+		{
+			$user_id = ANONYMOUS;
+		}
+		else
+		{
+			$user_ary = $this->ppde_operator->query_donor_user_data('username', $transaction_data['MT_USERNAME']);
+
+			if (!$user_ary)
+			{
+				$errors[] = $this->language->lang('PPDE_MT_DONOR_NOT_FOUND', $transaction_data['MT_USERNAME']);
+			}
+
+			$user_id = $user_ary['user_id'];
```

Comment:
If the entered username doesn't exist, `$user_ary` will be set to `false`. ou will not be able to find the index `user_id`